airbyte-integrations/connectors/source-gitlab/source_gitlab/source.py

```diff
@@ -69,7 +71,17 @@ def get_new_token_expiry_date(access_token_expires_in: int, access_token_created
         return pendulum.from_timestamp(access_token_created_at + access_token_expires_in)
 
     def refresh_access_token(self) -> Tuple[str, int, int, str]:
-        response_json = self._get_refresh_access_token_response()
+        try:
+            response_json = self._get_refresh_access_token_response()
+        except HTTPError as e:
+            content = e.response.json()
+            if e.response.status_code == 400 and content.get("error") == "invalid_grant":
+                raise AirbyteTracedException(
+                    internal_message=content.get("error_description"),
+                    message="Refresh token is invalid or expired. Please re-authenticate to restore access to Gitlab.",
```

Comment:
We should also guide the Customer where to do so, like "... Please re-authenticate from Sources > your source > settings"